entry_types/scrolled/package/src/frontend/MediaPlayer/index.js

```diff
@@ -0,0 +1,28 @@
+import {useEffect, useState, useRef} from 'react';
+import {media} from 'pageflow/frontend';
+
+export function MediaPlayer(props){
+  const playerWrapper = useRef(null);
+  const [player, setPlayer] = useState(undefined);
+  
+  useEffect( () => {    
+    if (props.sources && player == undefined) {
+      setPlayer(media.getPlayer(props.sources, {
+        poster: props.poster
+      }));
+    }
+  } );
+
+  return (
+    <div ref={playerWrapper}>
+      {renderPlayer(props, player, playerWrapper)}
+    </div>
+  );
+
+}
+
+function renderPlayer(props, player, parentElement) {
+  if (player && parentElement.current.childElementCount == 0) {
+    parentElement.current.appendChild(player.el());
+  }
+}
```

Comment:
Render functions should be side effect free. You can use a `useEffect` hook to obtain and append the player. Then you also do not need the state. Since this component manipulates the DOM, we should make sure that React  does not update the DOM which might erase our changes. You can take a look at the way `MediaTag` from `pageflow-paged-react` uses `shouldComponentUpdate` to this. For functional components, we can use `React.memo` to achieve a similar thing.